lib/buffer.js

```diff
@@ -176,12 +176,25 @@ Buffer.from = function(value, encodingOrOffset, length) {
   if (isAnyArrayBuffer(value))
     return fromArrayBuffer(value, encodingOrOffset, length);
 
+  if (value == null)
+    throw new TypeError(kFromErrorMsg);
+
+  if (typeof value === 'number')
+    throw new TypeError('"value" argument must not be a number');
+
+  const valueOf = value.valueOf();
```

Comment:
I wonder if we should be more cautious here or not, in case someone passes say `Object.create(null)`. Perhaps `value.valueOf && value.valueOf()` at the very least ?